lint-consul-retry.go

```diff
@@ -93,42 +90,49 @@ type visitor struct {
 	depth       int
 	currentTest string
 	path        string
+	retryDepth  int
+	requireNew  bool
 }
 
 func (v visitor) Visit(n ast.Node) ast.Visitor {
-	// Walk uses DFS so reset when we pop back up
-	if retryDepth > 0 && v.depth <= retryDepth {
-		retryDepth = 0
+	if n != nil {
+		// When called with a non-nil ast.Node, we are delving 1 node deeper into the tree
+		// and therefore should update our tracked depth accordingly.
+		v.depth++
+	} else {
+		// Once a sub-tree of the AST has finished being walked Visit(nil) will be invoked.
+		// There is no need to decrement the depth because we are passing the visitor by
+		// value rather than a reference. The previous visitor on up the stack with the correct
+		// depth would be used for subsequent recursive Visit calls.
```

Comment:
Thank you for adding those comments! It would have took me a while to understand this 😅